src/rotation.zig

```diff
@@ -320,6 +320,115 @@ pub const RotationModes = struct {
 			return true;
 		}
 	};
+	pub const Branch = struct { // MARK: Branch
+		pub const id: []const u8 = "branch";
+		pub const dependsOnNeighbors = true;
+		var branchModels: std.StringHashMap(u16) = undefined;
+		const BranchData = packed struct(u6) {
+			isConnectedNegX: bool,
+			isConnectedPosX: bool,
+			isConnectedNegY: bool,
+			isConnectedPosY: bool,
+			isConnectedNegZ: bool,
+			isConnectedPosZ: bool,
+		};
+
+		fn init() void {
+			branchModels = .init(main.globalAllocator.allocator);
+		}
+
+		fn deinit() void {
+			branchModels.deinit();
+		}
+
+		fn branchTransform(quad: *main.models.QuadInfo, data: BranchData) void {
+			for(&quad.corners, &quad.cornerUV) |*corner, *cornerUV| {
+				// X
+				if(!data.isConnectedNegX and corner[0] == 0) {
```

Comment:
Honestly, I have no idea what I am doing, but while paired with UV coordinates from `branch.obj` it seems to work, so IMO profit.